src/Features/Core/Portable/ExtractMethod/MethodExtractor.Analyzer.cs

```diff
@@ -91,7 +94,13 @@ public async Task<AnalyzerResult> AnalyzeAsync()
                 var thisParameterBeingRead = (IParameterSymbol)dataFlowAnalysisData.ReadInside.FirstOrDefault(s => IsThisParameter(s));
                 var isThisParameterWritten = dataFlowAnalysisData.WrittenInside.Any(s => IsThisParameter(s));
 
-                var instanceMemberIsUsed = thisParameterBeingRead != null || isThisParameterWritten;
+                // Checks to see if selection includes a non-static local function call + if the given local function declaration is not included in the selection.
+                var containsNonStaticLocalFunctionCallNotWithinSpan = symbolMap.Keys.Any(s => s.IsLocalFunction() && !s.IsStatic && !s.Locations.Any(l => SelectionResult.FinalSpan.Contains(l.SourceSpan)));
```

Comment:
nit: You're doing a very similar thing twice in the map. 

Could be simplified to 

```csharp
var localFunctionCallsNotWithinSpan = symbolMap.Keys.Where(s => s.IsLocalFunction() && !s.Locations.Any(l => SelectionResult.FinalSpan.Contains(l.SourceSpan)));
var containsAnyLocalFunctionCallNotWithinSpan = localFunctionCallsNotWithinSpan.Any();
var containsNonStaticLocalFunctionCallNotWithinSpan  = containsAnyLocalFunctionCallNotWithinSpan  && localFunctionCallsNotWithinSpan.Where(s => !s.IsStatic).Any();
```